sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Watch.java

```diff
@@ -559,24 +583,34 @@ public String toString(KV<FirstStateT, SecondStateT> state) {
     @Nullable
     abstract Coder<OutputT> getOutputCoder();
 
-    abstract Builder<InputT, OutputT> toBuilder();
+    abstract Builder<InputT, OutputT, KeyT> toBuilder();
 
     @AutoValue.Builder
-    abstract static class Builder<InputT, OutputT> {
-      abstract Builder<InputT, OutputT> setPollFn(Contextful<PollFn<InputT, OutputT>> pollFn);
+    abstract static class Builder<InputT, OutputT, KeyT> {
+      abstract Builder<InputT, OutputT, KeyT> setPollFn(Contextful<PollFn<InputT, OutputT>> pollFn);
+
+      abstract Builder<InputT, OutputT, KeyT> setOutputKeyFn(
+          @Nullable SerializableFunction<OutputT, KeyT> outputKeyFn);
 
-      abstract Builder<InputT, OutputT> setTerminationPerInput(
+      abstract Builder<InputT, OutputT, KeyT> setKeyCoder(Coder<KeyT> keyCoder);
+
+      abstract Builder<InputT, OutputT, KeyT> setTerminationPerInput(
           TerminationCondition<InputT, ?> terminationPerInput);
 
-      abstract Builder<InputT, OutputT> setPollInterval(Duration pollInterval);
+      abstract Builder<InputT, OutputT, KeyT> setPollInterval(Duration pollInterval);
+
+      abstract Builder<InputT, OutputT, KeyT> setOutputCoder(Coder<OutputT> outputCoder);
 
-      abstract Builder<InputT, OutputT> setOutputCoder(Coder<OutputT> outputCoder);
+      abstract Growth<InputT, OutputT, KeyT> build();
+    }
 
-      abstract Growth<InputT, OutputT> build();
+    /** Specifies the coder for the output key. */
+    public Growth<InputT, OutputT, KeyT> withKeyCoder(Coder<KeyT> keyCoder) {
```

Comment:
<!--new_thread; commit:1199ebf360c6a28d4e580be6c561bc1218726828; resolved:0-->
s/withKeyCoder/withOutputKeyCoder and s/keyCoder/outputKeyCoder ?

Reply:
Done.